departures.go

```diff
@@ -153,6 +171,51 @@ func main() {
 	}
 }
 
+func searchStations(name string) ([]station, error) {
+	var stations []station
+	err := getJSON(&stations, "https://2.bvg.transport.rest/locations?query=%s&poi=false&addresses=false", name)
+
+	return stations, err
+}
+
+func promptForStation(name string) (*station, error) {
+	stations, err := searchStations(name)
+	if err != nil {
+		return nil, fmt.Errorf("could not query stations")
+	}
+
+	if l := len(stations); l == 0 {
+		// no stations found
+		return nil, fmt.Errorf("could not find matching stations")
+	} else if l == 1 {
+		return &stations[0], nil
+	}
+	// set first result as fallback
+	fallback := stations[0].Name
+
+	// convert to map[string]station to get station after user prompt
+	var options []string
+	optionStation := map[string]*station{}
+	for _, s := range stations {
+		options = append(options, s.Name)
+		optionStation[s.Name] = &s
```

Comment:
This results in all the entries pointing to the same station (because `s` is overwritten in enery loop).

Reply:
I would suggest to decode into a `[]*station` and to just assign `optionStation[s.Name] = s` here.